manifests/pub_dir.pp

```diff
@@ -23,18 +23,10 @@
       content     => template('foreman_proxy_content/httpd_pub.erb'),
       ssl_content => template('foreman_proxy_content/httpd_pub.erb'),
     }
-  } elsif $foreman_proxy_content::pulp {
-    pulp::apache::fragment { 'pub_dir':
-      ssl_content => template('foreman_proxy_content/httpd_pub.erb'),
-    }
-
-    file { "${apache::confd_dir}/pulp-vhosts80/pub_dir.conf":
-      ensure  => file,
-      content => template('foreman_proxy_content/httpd_pub.erb'),
-      owner   => 'apache',
-      group   => 'apache',
-      mode    => '0600',
-      notify  => Service['httpd'],
+  } elsif $foreman_proxy_content::pulpcore_mirror {
+    pulpcore::apache::fragment{ 'pub_dir':
```

Comment:
Logically speaking the else branch is dead code now. `$shared_with_foreman_vhost` is dfined as `!$pulpcore_mirror` so you end up with:
```puppet
if !$x {
} elsif $x {
} else {
}
```

Given that `pulpcore::apache::fragment` does the right thing even if included in a shared config with Foreman, it may be sufficient to only have a pulpcore fragment.